roles/servicetelemetry/tasks/component_servicemonitor.yml

```diff
@@ -27,14 +27,26 @@
               - action: labeldrop
                 regex: publisher
                 sourcelabels: []
-            port: prom-http
+            port: prom-https
+            scheme: https
+            tlsConfig:
+              caFile: /etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt
+              serverName: "{{ this_smartgateway }}.{{ meta.namespace }}.svc"
+            bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
         selector:
           matchLabels:
             app: smart-gateway
-  when: servicemonitor_manifest is not defined
```

Comment:
This is a bit of a change in how the `servicemonitor_manifest` override works, and I wasn't sure the best way to implement it.

Previously a single serviceMonitor would cover all the SGs, and we support overriding that via `servicemonitor_manifest`. Now, there is a unique serviceMonitor required for each SG, and instead of making it either/or, this code generates the required ones unconditionally and adds an additional manifest if you've provided one. 

Thoughts?

Reply:
Yea this isn't the first time this has kinda bitten us since the move to creating multiple SGs.

I think your change here actually helps this situation somewhat though. I would now make it so you can define your SG via the `clouds` parameter, and perhaps move `servicemonitorManifest` to that object, so you can do an override per-SG (which is a desired feature; Paul and I were just talking about this the other day for testing something).

I think the `when` would now basically look for a non-empty (defined) `servicemonitor_manifest` which I think would be called something like `this_collector.servicemonitor_manifest`. We'd just drop the `servicemonitor_manifest` from the top-level.

Does that make sense?

Reply:
> Actually I'm pulling my recommendation out for this. I recently opened this issue which I think is a better solution to custom manifest management vs inlining everything: https://bugzilla.redhat.com/show_bug.cgi?id=2024992

Upon further investigation I still think the solution to dealing with the outstanding `servicemonitorManifest` that is more global is one of the following two approaches:

* drop servicemonitorManifest as an option entirely and document as such
* implement this on a per-cloud basis as described above

I was looking at some other work on Friday for scaling SGs that I wanted to make sure wouldn't just immediately deprecate the 2nd bullet approach should it be implemented, but upon further review I think the implementation will greatly benefit from the changes here, and not conflict should the per-cloud implementation logic be implemented.

Reply:
I like the per-cloud servicemonitor override idea, but would be just as happy leaving it for a separate PR as I was trying not to alter the CRD interface at all. Do you think I should roll it into this one?

Reply:
I think this is fine as-is, and we can roll that feature into a different PR. What I would do though is create a BZ and a release note calling out how `servicemonitorManifest` works differently now in STF 1.4 for anyone upgrading from STF 1.3 to 1.4 who happens to be using that feature.

That should cover us until we change the way this works.

Reply:
The BZ that technically spawned this change is about user logins to dashboards, so a release note about servicemonitor manifests seemed out of place. I've opened https://bugzilla.redhat.com/show_bug.cgi?id=2026005 and asked Joanne if I did it right. Either way, it's "in the system" (or two!) so we shouldn't forget about it.